### PR TITLE
ref(ts): Fix lint issues that are being masked

### PR DESCRIPTION
**Pull request by billyvg**
These are lint issues that will come up in a new version of `@typescript-eslint`. See https://github.com/getsentry/eslint-config-sentry/pull/105

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 !.github
 !.github/workflows/scripts/*
 !.storybook
+*.d.ts
```
Comment (billyvg) [inline .eslintignore:10]: No need to lint this

File: static/app/components/commitLink.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ type CommitFormatterParameters = {
 type CommitProvider = {
   icon: React.ReactNode;
   providerIds: string[];
-  commitUrl: (CommitFormatterParameters) => string;
+  commitUrl: (opts: CommitFormatterParameters) => string;
```
Comment (billyvg) [inline static/app/components/commitLink.tsx:18]: Wrong syntax here

File: static/app/components/contextData/index.tsx [hunk 3]
```diff
@@ -29,6 +29,10 @@ type State = {
   withAnnotatedText: boolean;
 };
 
+function getValueWithAnnotatedText(v: Value, meta?: Meta) {
+  return <AnnotatedText value={v} meta={meta} />;
+}
```
Comment (billyvg) [inline static/app/components/contextData/index.tsx:34]: I pulled this out because of `no-shadow`

File: static/app/utils/theme.tsx [hunk 4]
```diff
@@ -721,6 +721,7 @@ type MyTheme = Theme;
  * Configure Emotion to use our theme
  */
 declare module '@emotion/react' {
+  // eslint-disable-next-line @typescript-eslint/no-shadow
```
Comment (billyvg) [inline static/app/utils/theme.tsx:724]: This only errors in the new version of `@typescript-eslint`

File: static/app/utils/withExperiment.tsx [hunk 5]
```diff
@@ -100,8 +100,8 @@ function withExperiment<
       }
     }
 
-    getProps<P extends ExperimentType>() {
-      return this.props as unknown as ExpectedProps<P>;
+    getProps<Q extends ExperimentType>() {
+      return this.props as unknown as ExpectedProps<Q>;
```
Comment (billyvg) [inline static/app/utils/withExperiment.tsx:104]: `no-shadow`

File: static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx [hunk 6]
```diff
@@ -132,7 +132,7 @@ class AddCodeOwnerModal extends Component<Props, State> {
     const {codeMappings} = this.props;
     const codeMapping = codeMappings.find(mapping => mapping.id === codeMappingId);
     const {integrationId, provider} = codeMapping as RepositoryProjectPathConfig;
-    const errActors = errorJSON?.raw?.[0].split('\n').map(el => <p>{el}</p>);
+    const errActors = errorJSON?.raw?.[0].split('\n').map((el, i) => <p key={i}>{el}</p>);
```
Comment (billyvg) [inline static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx:135]: Not sure why this wasn't erroring... not great

**Review by evanpurkhiser — APPROVED** [review]

**Review by dashed — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**